src/ShopBundle/objects/TCMSSmartURLHandler/TCMSSmartURLHandler_ShopIPayment.class.php

```diff
@@ -13,7 +13,10 @@
 
 /**
  * assumes the path in the TCMSSmartURLData is a simple tree path.
-/**/
+ *
+ * @psalm-suppress UndefinedPropertyAssignment
+ * @FIXME Writing data into `$OURLData` when there is no magic `__set` method for them defined.
```

Comment:
In this class, there is a lot of writing into 'magic' properties of `TCMSSmartURLData` - however, those properties are not writable.